examples/move/token/sources/regulated_coin.move

```diff
@@ -0,0 +1,318 @@
+
+        let token = test::mint(3001_000000, ctx);
+        let request = token::transfer(token, ALICE, ctx);
+
+        limiter::verify(&policy, &mut request, ctx);
+
+        abort 1337
+    }
+
+    #[test]
+    /// Turn 1000 REG into Coin from.
+    fun test_limiter_to_coin_allowed_pass() {
+        let ctx = &mut test::ctx();
+        let (policy, cap) = policy_with_allowlist(ctx);
+
+        let token = test::mint(1000_000000, ctx);
+        let (coin, request) = token::to_coin(token, ctx);
+
+        limiter::verify(&policy, &mut request, ctx);
+        denylist::verify(&policy, &mut request, ctx);
+        allowlist::verify(&policy, &mut request, ctx);
+
+        token::confirm_request(&mut policy, request, ctx);
+        test::return_policy(policy, cap);
+        coin::burn_for_testing(coin);
+    }
+
+    #[test, expected_failure(abort_code = limiter::ELimitExceeded)]
+    /// Try to convert more than 1000.00 REG in a single operation.
+    fun test_limiter_to_coin_exceeded_fail() {
+        let ctx = &mut test::ctx();
+        let (policy, _cap) = policy_with_allowlist(ctx);
+
+        let token = test::mint(1001_000000, ctx);
+        let (_coin, request) = token::to_coin(token, ctx);
+
+        limiter::verify(&policy, &mut request, ctx);
+
+        abort 1337
+    }
+
+    // === Allowlist Tests ===
+
+    // Test from allowed account is already covered in the
+    // `test_limiter_transfer_allowed_pass`
+
+    #[test, expected_failure(abort_code = allowlist::ENotAllowed)]
+    /// Try to `transfer` to a not allowed account.
+    fun test_allowlist_transfer_to_not_allowed_fail() {
+        let ctx = &mut test::ctx();
+        let (policy, _cap) = policy_with_allowlist(ctx);
+
+        let token = test::mint(1000_000000, ctx);
+        let request = token::transfer(token, BOB, ctx);
+
+        allowlist::verify(&policy, &mut request, ctx);
+
+        abort 1337
+    }
+
+    #[test, expected_failure(abort_code = allowlist::ENotAllowed)]
+    /// Try to `from_coin` from a not allowed account.
+    fun test_allowlist_from_coin_not_allowed_fail() {
+        let ctx = &mut test::ctx();
+        let (policy, cap) = test::get_policy(ctx);
+
+        set_rules(&mut policy, &cap, ctx);
+
+        let coin = coin::mint_for_testing(1000_000000, ctx);
+        let (_token, request) = token::from_coin(coin, ctx);
+
+        allowlist::verify(&policy, &mut request, ctx);
+
+        abort 1337
+    }
+
+    // === Denylist Tests ===
+
+    #[test, expected_failure(abort_code = denylist::EUserBlocked)]
+    /// Try to `transfer` from a blocked account.
+    fun test_denylist_transfer_fail() {
+        let ctx = &mut test::ctx();
+        let (policy, _cap) = policy_with_denylist(ctx);
+
+        let token = test::mint(1000_000000, ctx);
+        let request = token::transfer(token, BOB, ctx);
+
+        denylist::verify(&policy, &mut request, ctx);
+
+        abort 1337
+    }
+
+    #[test, expected_failure(abort_code = denylist::EUserBlocked)]
```

Comment:
Do you want it to be identical?

Reply:
Identical would be fine because it's prefixed by the package name.  Exactly opposite would be fine as well, e.g. `denylist::EUserDenied`, `allowlist::EUserNotAllowed`, but the two similar terms that weren't the same was confusing to me.